src/pages/RegPage.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { useReducer, useState } from "react";
+import DisplayRequest from "../views/DisplayRequest";
+import FlowData from "../views/FlowData";
+import API, { getFormattedURL, getFormattedProxyURL } from "../API";
+import RegistrationProvider from "../util/RegistrationProvider";
+import { Container, Row, Col } from "react-bootstrap";
+import Registration from "../views/Registration";
+
+const RegPage = (props) => {
+    const reset = () => {
+        setRequest({});
+        setResponse({});
+        setCSRF("");
+        setFlowID("");
+        setSessionID("");
+    }
+
+    const resetPrint = () => {
+        setRequest({});
+        setResponse({});
+    }
+
+    const [request, setRequest] = useState({});
+
+    const [response, setResponse] = useState({});
+
+    const [flowID, setFlowID] = useState("");
+    const [csrf, setCSRF] = useState("");
+    const [accountKey, setAccountKey] = useState("");
+    const [sessionID, setSessionID] = useState("");
+
```

Comment:
```suggestion

const registration = new RegistrationProvider(props.cluster, props.flow);

```

The Registration provider depends on properties. It will be recreated anyway, if the props update. So we can create one on component rendering.